executor/runtime/docker/docker.go

```diff
@@ -558,8 +558,36 @@ func sleepWithCtx(parentCtx context.Context, d time.Duration) error {
 	}
 }
 
-func (r *DockerRuntime) dockerPull(ctx context.Context, c *runtimeTypes.Container) error {
-	return pullWithRetries(ctx, r.metrics, r.client, c, doDockerPull)
+func imageExists(ctx context.Context, client *docker.Client, ref string) (bool, error) {
+	resp, _, err := client.ImageInspectWithRaw(ctx, ref)
+	if err != nil {
+		if docker.IsErrImageNotFound(err) {
+			return false, nil
+		}
+
+		return false, err
+	}
+
+	log.Debugf("Image '%s' exists. Response: %+v", ref, resp)
+	return true, nil
+}
+
+// DockerPull returns true if the image was cached, and we didn't need to pull
+func (r *DockerRuntime) DockerPull(ctx context.Context, c *runtimeTypes.Container) (bool, error) {
+	imgName := c.QualifiedImageName()
+	imgExists, err := imageExists(ctx, r.client, imgName)
+	if err != nil {
+		log.Warningf("Error inspecting image '%s', due to reason: %+v", imgName, err)
```

Comment:
So, this might be worth logging at Error.

One of the neat features of out logging library is it supports structured fields. 

For arbitrary string fields:
https://godoc.org/github.com/sirupsen/logrus#WithField

For error fields:
https://godoc.org/github.com/sirupsen/logrus#WithError